vm/src/stdlib/collections.rs

```diff
@@ -240,26 +241,44 @@ mod _collections {
             Ok(())
         }
 
+        fn adjust_negative_index(&self, index: i32) -> usize {
+            if index < 0 {
+                max(index + self.borrow_deque().len() as i32, 0) as usize
+            } else {
+                index as usize
+            }
+        }
+
         #[pymethod]
         fn index(
             &self,
             obj: PyObjectRef,
-            start: OptionalArg<usize>,
-            stop: OptionalArg<usize>,
+            start: OptionalArg<i32>,
```

Comment:
Don't have time to do a full review yet but `i32`? Indices usually take values in the range supplied by `isize`.

Reply:
ah.. i didn't know it. i would fix that